File: zig/src/primes.zig
const std = @import("std");

fn isPrime(n: usize) bool {
    const limit = std.math.sqrt(n);
    var d: usize = 2;
    while (d <= limit) : (d += 1) {
        if (n % d == 0) return false;
    }
    return true;
}

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();

    var primes = std.ArrayList(usize).init(allocator);
    defer primes.deinit();

    for (2..101) |n| {
        if (isPrime(n)) try primes.append(n);
    }

    std.debug.print("{any}\n", .{primes.items});
}

File: typescript/src/primes.ts
const range = (start: number, end: number) =>
  Array.from({ length: end - start + 1 }, (_, i) => start + i)

const primes = range(2, 100).filter(n =>
  range(2, Math.floor(Math.sqrt(n))).every(d => n % d !== 0)
)

console.log(primes)
